site/docs/plugins/chat-members.md

```diff
@@ -5,18 +5,280 @@ next: false
 
 # Chat Members Plugin (`chat-members`)
 
-Automatically store information about users in a chat and retrieve it easily.
-Track group and channel members, and list them.
+Telegram doesn't offer a method in the Bot API to retrieve the members of a chat, you have to keep track of them yourself.
+This plugin makes it easy to work with `ChatMember` objects, by offering a convenient way to listen for changes in the form of custom filters, and by storing and updating the objects.
 
 ## Introduction
 
-In many situations, it is necessary for a bot to have information about all the users of a given chat.
-Currently, though, the Telegram Bot API exposes no method that allows us to retrieve this information.
+Working with `ChatMember` objects from the Telegram Bot API can sometimes be cumbersome.
+There are several different statuses that are often interchangeable in most applications.
+In addition, the restricted status is ambiguous because it can represent both members of the group and restricted users that are not in the group.
 
-This plugin comes to the rescue: automatically listening to `chat_member` events and storing all `ChatMember` objects.
+This plugin simplifies dealing with chat members by offering fully typed filters for chat member updates.
 
 ## Usage
 
+### Chat Member Filters
+
+You can listen for two kinds of updates regarding chat members using a Telegram bot: `chat_member` and `my_chat_member`.
+Both of them specify the old and new status of the user.
+
+- `my_chat_member` updates are received by your bot by default and they inform you about the status of the bot being updated in any chat, as well as users blocking the bot;
+- `chat_member` updates are only received if you explicitly include them in the list of allowed updates, they notify about any status changes for users in chats **where your bot is admin**.
+
+Instead of manually filtering the old and new status, chat member filters do this automatically for you, allowing you to react to every type of transition you're interested in.
```

Comment:
```suggestion
Instead of manually filtering the old and the new statuses, chat member filters do this automatically for you, allowing you to act on any type of transition you're interested in.
```